### PR TITLE
Payments - modify link to open new payment plan modal

### PR DESCRIPTION
**Pull request by lukaszwilczakwilnet**
#### Changes proposed in this Pull Request:
- for payments blocks add hash "add-new-payment-plan" to link to new payment plan. This hash allow to open modal modal on page load.

Fixes # https://github.com/Automattic/wp-calypso/issues/62888

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
Checkout the PR
Login to admin panel
Choose paid plan to have access to payments blocks
Choose theme with widget areas (eg. twenty twenty one)
Go to Customizer
Go to Widgets, select any widget area
Add new payment block
Browse for current plans and click on the bottom to "Add a new payment plan"
Check if you are redirected to /earn/payment-plans#add-new-payment-plan, not just add-new-payment-plan

link-hash

### CODE REVIEW

**Review by dsas — DISMISSED** [review]
works well 👍

File: projects/plugins/jetpack/changelog/update-new-plan-open-in-modal [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: enhancement
+
+Open add new subscription modal when there's hash in URL
```
Comment (dsas) [inline projects/plugins/jetpack/changelog/update-new-plan-open-in-modal:4]: This changelog entry is misleading though, this change puts the hash in the url, not opens a new subscription modal.

**Comment by BogdanUngureanu** [comment]
@lukaszwilczakwilnet  looks like the PR is not up to date. Can you also merge the master branch into this branch? :)

**Review by jeherve — APPROVED** [review]

**Comment by Copons** [comment]
r247172-wpcom

**Pull request merged**